crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -526,10 +526,12 @@ impl Acir {
                 }
                 _ => {
                     if self.arith_cache.contains_key(a) {
-                        if let Some(w) = self.arith_cache[a].clone().witness {
+                        let var = self.arith_cache[a].clone();
+                        if let Some(w) = var.witness {
```

Comment:
Not a big deal, but if you think the following is more readable:
```
let witness = if let Some(w) = var.witness { w} else {generate_witness(&var, evaluator)};
inputs.push(GadgetInput { witness, num_bits: l_obj.size_in_bits() });
```

Reply:
I changed to using `unwrap_or_else`